data/migrations/deb/1.3.396_to_1.3.397.py

```diff
@@ -0,0 +1,197 @@
+                         .format(db_name, os.path.join(leveldb_ledger_dir, db_name)))
+            return False
+
+    # Migrate other storages with non-integer keys
+    for db_name in ["attr_db", "idr_cache_db", "seq_no_db", "state_signature"]:
+        if not migrate_storage(leveldb_ledger_dir, rocksdb_ledger_dir, db_name, False):
+            logger.error("Could not migrate {}, DB path: {}"
+                         .format(db_name, os.path.join(leveldb_ledger_dir, db_name)))
+            return False
+
+    return True
+
+
+def migrate_all():
+    node_name = get_node_name()
+    if node_name is None:
+        logger.error("Could not get node name")
+        return False
+
+    config = getConfig()
+    config_helper = NodeConfigHelper(node_name, config)
+
+    leveldb_ledger_dir = config_helper.ledger_dir
+    rocksdb_ledger_dir = os.path.join(config_helper.ledger_data_dir, node_name + "_rocksdb")
+    if os.path.exists(rocksdb_ledger_dir):
+        logger.error("Temporary directory for RocksDB-based ledger exists, please remove: {}"
+                     .format(rocksdb_ledger_dir))
+        return False
+
+    try:
+        os.mkdir(rocksdb_ledger_dir)
+    except Exception:
+        logger.error(traceback.print_exc())
+        logger.error("Could not create temporary directory for RocksDB-based ledger: {}"
+                     .format(rocksdb_ledger_dir))
+        return False
+
+    logger.info("Starting migration of storages from LevelDB to RocksDB...")
+
+    if migrate_storages(leveldb_ledger_dir, rocksdb_ledger_dir):
+        logger.info("All storages migrated successfully from LevelDB to RocksDB")
+    else:
+        logger.error("Storages migration from LevelDB to RocksDB failed!")
+        shutil.rmtree(rocksdb_ledger_dir)
+        return False
+
+    # Archiving LevelDB-based ledger
+    try:
+        archive_leveldb_ledger(node_name, leveldb_ledger_dir)
+    except Exception:
+        logger.warning("Could not create an archive of LevelDB-based ledger, proceed anyway")
+
+    # TODO: it whould be nice to open new RocksDB-based ledger
+    # and compare root hashes with LevelDB-based ledger here
+
+    # Remove LevelDB-based ledger
+    try:
+        shutil.rmtree(leveldb_ledger_dir)
+    except Exception:
+        logger.error(traceback.print_exc())
+        logger.error("Could not remove LevelDB-based ledger: {}"
+                     .format(leveldb_ledger_dir))
+        shutil.rmtree(rocksdb_ledger_dir)
+        return False
+
+    ledger_dir = leveldb_ledger_dir
+
+    try:
+        shutil.move(rocksdb_ledger_dir, ledger_dir)
+    except Exception:
+        logger.error(traceback.print_exc())
+        logger.error("Could not rename temporary RocksDB-based ledger from '{}' to '{}'"
+                     .format(rocksdb_ledger_dir, ledger_dir))
+        shutil.rmtree(rocksdb_ledger_dir)
+        return False
+
+    set_own_perm("indy", ledger_dir)
+
+    return True
+
+
+if migrate_all():
```

Comment:
as far as I remember one of the requirements for the migration script - it should do nothing and finishes without any error if it runs several times
Also we have reinstall command that should run smoothly as well.
I think we need to determine if ledgers are leveldb or rocksdb somehow to achieve the reinstall/rerun goal.
Probably the easiest way is to add some sign to the name of the directory or something similar.